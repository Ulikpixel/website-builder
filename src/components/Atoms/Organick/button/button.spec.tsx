import React from 'react'
import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import Button, { variants, VariantsType } from './button'

const validVariant = (variant: VariantsType | undefined = 'blue-dark'): void => {
  it(`renders with default "${variant}" variant and children`, () => {
    const { getByText } = render(<Button variant={variant}>Click me</Button>)
    const button = getByText('Click me')
    const imageElement = screen.queryByRole('img')

    expect(button).toBeInTheDocument()
    expect(button).toHaveClass(variants[variant])
    expect(imageElement).toBeNull()
    expect(button).toMatchSnapshot()
  })
}

describe('Button component', () => {
  validVariant()
  validVariant('white')
  validVariant('yellow')

  it('renders with icon when icon prop is true', () => {
    const { getByText } = render(<Button icon>Click me</Button>)
    const imageElement = screen.getByRole('img')
    const button = getByText('Click me')

    expect(imageElement).toBeInTheDocument()
    expect(button).toMatchSnapshot()
  })
})
