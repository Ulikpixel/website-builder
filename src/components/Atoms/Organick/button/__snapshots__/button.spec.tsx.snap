// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders with default "blue-dark" variant and children 1`] = `
<a
  class="
        rounded-xl font-roboto font-bold transition-colors gap-2 items-center
        px-6 py-4 text-xl sm:py-5 lg:px-10 inline-flex
       
    bg-organick-blue-dark text-default-white border-organick-blue-dark border
    hover:bg-default-white hover:text-organick-blue-dark
  "
  href="/"
>
  Click me
</a>
`;

exports[`Button component renders with default "white" variant and children 1`] = `
<a
  class="
        rounded-xl font-roboto font-bold transition-colors gap-2 items-center
        px-6 py-4 text-xl sm:py-5 lg:px-10 inline-flex
       
    bg-organick-white border-organick-blue-dark border text-organick-blue-dark
    hover:bg-organick-blue-dark hover:text-default-white
  "
  href="/"
>
  Click me
</a>
`;

exports[`Button component renders with default "yellow" variant and children 1`] = `
<a
  class="
        rounded-xl font-roboto font-bold transition-colors gap-2 items-center
        px-6 py-4 text-xl sm:py-5 lg:px-10 inline-flex
       
    bg-organick-yellow text-organick-blue-dark border-organick-yellow border
    hover:bg-default-white
  "
  href="/"
>
  Click me
</a>
`;

exports[`Button component renders with icon when icon prop is true 1`] = `
<a
  class="
        rounded-xl font-roboto font-bold transition-colors gap-2 items-center
        px-6 py-4 text-xl sm:py-5 lg:px-10 inline-flex
       
    bg-organick-blue-dark text-default-white border-organick-blue-dark border
    hover:bg-default-white hover:text-organick-blue-dark
  "
  href="/"
>
  Click me
  <img
    alt="arrow icon"
    class="w-4 h-4 relative top-[0.3px]"
    data-nimg="1"
    decoding="async"
    height="40"
    loading="lazy"
    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
    style="color: transparent;"
    width="40"
  />
</a>
`;
